### PR TITLE
Release v1.1.0

### LINKED ISSUE #4: Remove hardcoded settings
Several hardcoded setting names and values are present in `confluent/confluent.go`. The names should be extracted to constants, and the values should should be extracted as default values to be loaded if client configuration doesn't specify a value.

### PR DESCRIPTION
**Pull request by davejoyce**
Release version **v1.1.0** changes
* Removed hardcoded configuration settings
* Changed `confluent/confluent.go` functions to
  * use extracted interfaces rather than struct types from [confluent-kafka-go](https://github.com/confluentinc/confluent-kafka-go) for improved testability
  * return error rather than panic

### CODE REVIEW

File: confluent/confluent.go [hunk 1]
```diff
@@ -57,13 +106,35 @@ func createBasicConfigMap(properties map[string]string) kafka.ConfigMap {
...
+// CreateTopic creates a topic using the Admin Client API with default settings.
+// This function returns an error if the topic does not exist and cannot be
+// created for any reason.
+func CreateTopic(producer k.Producer, topic string) error {
+	topics := []string{topic}
+
+	topicCreationConfig := make(map[string]string)
+	topicCreationConfig[AdminOperationTimeout] = DefaultAdminOperationTimeout
+	topicCreationConfig[NumPartitions] = DefaultNumPartitions
+	topicCreationConfig[ReplicationFactor] = DefaultReplicationFactor
```
Comment (hudcoley) [inline confluent/confluent.go:118]: Do we want to make these configurable, or will they always be set to the default?

Comment (davejoyce) [reply confluent/confluent.go:118]: The 1 topic `CreateTopic` method is a convenience method that delegates to the fuller `CreateTopics` method, using defaults. If you want to specify your configuration, you can use the `CreateTopics` method directly.

Comment (mknutty2) [reply confluent/confluent.go:118]: i suggest adding something for retention as that is  very important and must be determined upfront. I am not sure if  kafka has a default.

Comment (hudcoley) [reply confluent/confluent.go:118]: that makes sense. thanks for the clarification

**Review by hudcoley — APPROVED** [review]
Looks good to me. I like the abstraction to interfaces and then the mocks for testing. Will make this really easy to test

**Review by mknutty2 — APPROVED** [review]

**Pull request merged**